### PR TITLE
[feature] caching of offset in DateTime moment#506

### PR DESCRIPTION
**Pull request by neversaid**
This will avoid fetching unnecessary often from zone.offset - we already set the offset in the constructor to the private property o on DateTime.

### CODE REVIEW

**Comment by neversaid** [comment]
Thx!
Jepp, honestly i am not that happy with the current constructor in regards of readability :).
But i already had a peak to the v2 - if you want me to help for v2 (doing most of the time typescript ;) ) let me know.

File: src/datetime.js [hunk 1]
```diff
@@ -1114,7 +1114,7 @@ export default class DateTime {
    * @type {number}
    */
   get offset() {
-    return this.isValid ? this.zone.offset(this.ts) : NaN;
```
Comment (icambron) [inline src/datetime.js]: Why does this need a `+`? Seems like it would already be a number?

Comment (neversaid) [reply src/datetime.js]: but i can be null as well if in some crazy case the invalid is false...

**Comment by GillesDebunne** [comment]
> But i already had a peak to the v2 - if you want me to help for v2 (doing most of the time typescript ;) ) let me know.

Sure, feel free to jump in, any comment on the code is welcome!

**Comment by icambron** [comment]
Thanks for the contribution!

**Pull request merged**